ui-tests/src/test/java/io/syndesis/qe/steps/integrations/editor/EditorSteps.java

```diff
@@ -0,0 +1,164 @@
+
+    @Then("^she is presented with a visual integration editor$")
+    public void verifyNewIntegrationEditorOpened() {
+        editor.getRootElement().shouldBe(visible);
+        chooseConnection.getRootElement().shouldBe(visible);
+        flowViewComponent.getRootElement().shouldBe(visible);
+    }
+
+    @Then("^she is presented with a visual integration editor for \"([^\"]*)\"$")
+    public void verifyEditorOpenedFor(String integrationName) {
+        this.verifyNewIntegrationEditorOpened();
+        log.info("editor must display integration name {}", integrationName);
+        assertThat(flowViewComponent.getIntegrationName(), is(integrationName));
+    }
+
+    @When("^she adds first step between START and STEP connection$")
+    public void sheAddsFirstStep() throws Throwable {
+        flowViewComponent.clickAddStepLink(0);
+    }
+
+    @When("^she adds second step between STEP and FINISH connection$")
+    public void sheAddsSecond() throws Throwable {
+        flowViewComponent.clickAddStepLink(1);
+    }
+
+    @When("^.*adds? integration \"([^\"]*)\" on position \"([^\"]*)\"$")
+    public void addAnotherStep(String type, int stepPos) throws Throwable {
+        if (type.equalsIgnoreCase("connection")) {
+            flowViewComponent.clickAddConnectionLink(stepPos);
+        } else {
+            flowViewComponent.clickAddStepLink(stepPos);
+        }
+    }
+
+    /**
+     *
+     * unused
+     */
+    @And("^.*checks? that text \"([^\"]*)\" is \"([^\"]*)\" in hover table over \"([^\"]*)\" step$")
+    public void checkTextInHoverTable(String text, String isVisible, String stepPosition) throws InterruptedException {
+        if (isVisible.equalsIgnoreCase("visible")) {
+            Assertions.assertThat(flowViewComponent.checkTextInHoverTable(stepPosition))
+                    .isNotEmpty()
+                    .containsIgnoringCase(text);
+        } else {
+            Assertions.assertThat(flowViewComponent.checkTextInHoverTable(stepPosition))
+                    .isNotEmpty()
+                    .doesNotContain(text);
+        }
+    }
+
+    /**
+     * Every step element has class step and every option to add step/connection also has class step.
+     * So if you have 3 steps created, stepPosition is: first = 0, second = 2, third = 4 etc.
+     *
+     * @param text
+     * @param isVisible
+     * @param stepPosition
+     * @throws InterruptedException
+     */
+    @And("^.*checks? that text \"([^\"]*)\" is \"([^\"]*)\" in step warning inside of step number \"([^\"]*)\"$")
+    public void checkTextInStepWarning(String text, String isVisible, int stepPosition) throws InterruptedException {
+        if (isVisible.equalsIgnoreCase("visible")) {
+            doCheckTextInStepsWarningTable(text, stepPosition, true);
+        } else {
+            doCheckTextInStepsWarningTable(text, stepPosition, false);
+        }
+
+    }
+
+    /**
+     *
+     * ??????????????????????????????????????????????????
```

Comment:
There is warning with text when something is wrong with the integration like missing datamapper step

Reply:
This is a remnant of the process of moving steps to multiple step classes. We have talked about that. But thanks for the explanation.